pkg/test/utils/kubernetes.go

```diff
@@ -1217,6 +1219,76 @@ func RunAssertCommands(ctx context.Context, logger Logger, namespace string, com
 	return RunCommands(ctx, logger, namespace, convertAssertCommand(commands, timeout), workdir, timeout, kubeconfigOverride)
 }
 
+// RunAssertExpressions evaluates a set of CEL expressions specified as AnyAllExpressions
+func RunAssertExpressions(
+	ctx context.Context,
+	cl client.Client,
+	programs map[string]cel.Program,
+	resourceRefs []harness.TestResourceRef,
+	assertAny,
+	assertAll []*harness.Assertion,
+) []error {
+	errs := []error{}
+	if len(assertAny) == 0 && len(assertAll) == 0 {
+		return errs
+	}
+
+	variables := make(map[string]interface{})
+	for _, resourceRef := range resourceRefs {
+		namespacedName, referencedResource := resourceRef.BuildResourceReference()
```

Comment:
Thinking more about the namespace, this should probably default to the current test's namespace if not specified in the assert file 🤔